app.js

```diff
@@ -1,409 +1,81 @@
- * @param {string} [variable.defaultValue] - The default value of the variable (optional).
- * @returns {string} The location template string.
- */
-const generateLocationTemplate = ({ uri, value, source, defaultValue }) => {
-  return `
-    <span resource="${uri}" typeof="https://data.vlaanderen.be/ns/mobiliteit#Variabele">
-      <span property="dct:type" content="location"></span>
-      <span property="dct:source" resource="${source}"></span>
-      <span class="mark-highlight-manual" property="rdf:value">\${${value}}</span>
-      ${
-        !defaultValue?.length
-          ? ""
-          : `<span property="https://data.vlaanderen.be/ns/mobiliteit#standaardwaarde">${defaultValue}</span>`
-      }
-    </span>
-  `;
-};
-
-/**
- * Generates a date template.
- *
- * @param {Object} variable - The object containing the variable data.
- * @param {string} variable.uri - The URI of the variable.
- * @param {string} variable.value - The name of the variable.
- * @param {string} [variable.defaultValue] - The default value of the variable (optional).
- * @returns {string} The date template string.
- */
-const generateDateTemplate = ({ uri, value, defaultValue }) => {
-  return `
-    <span resource="${uri}" typeof="https://data.vlaanderen.be/ns/mobiliteit#Variabele">
-      <span property="dct:type" content="date"></span>
-      <span class="mark-highlight-manual" property="rdf:value" datatype="xsd:date">\${${value}}</span>
-      ${
-        !defaultValue?.length
-          ? ""
-          : `<span property="https://data.vlaanderen.be/ns/mobiliteit#standaardwaarde" datatype="xsd:date">${defaultValue}</span>`
-      }
-    </span>
-  `;
-};
-
-/**
- * Applies the mappings to the template and generates the annotated template.
- *
- * @param {string} basicTemplate - The basic template string.
- * @param {TemplateVariable[]} variables - The variables to be mapped.
- * @returns {string} The annotated template string.
- */
-export const applyTemplateMappings = (basicTemplate, variables) => {
-  let annotatedTemplate = basicTemplate;
-
-  const templateGenerators = {
-    codelist: (variable) =>
-      generateCodelistTemplate({
-        ...variable,
-        source: SPARQL_ENDPOINT,
-      }),
-    location: (variable) =>
-      generateLocationTemplate({
-        ...variable,
-        source: SPARQL_ENDPOINT,
-      }),
-    date: generateDateTemplate,
-    default: generateTextTemplate,
-  };
-  for (const variable of variables) {
-    if (variable.type === "instruction") continue;
 
-    const regex = new RegExp(`\\\${${variable.value}}`, "g");
-    const generator =
-      templateGenerators[variable.type] || templateGenerators.default;
-
-    annotatedTemplate = annotatedTemplate.replace(regex, generator(variable));
+  const updatedUris = extractInsertUris(req.body);
+  const templates = await getTemplatesAndVariables(updatedUris);
+  if (!templates.results.bindings.length) {
+    return res.send("No templates found");
```

Comment:
maybe we should just console.log it as we answer to the delta notifier and it doesn't care